### PR TITLE
[clang][dataflow] Fix a missing break from a switch case -Wimplicit-fallthrough

### PR DESCRIPTION
**Pull request by jvoung**
Missed when changing code in https://github.com/llvm/llvm-project/pull/120102

### CODE REVIEW

File: clang/lib/Analysis/FlowSensitive/SmartPointerAccessorCaching.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ bool hasSmartPointerClassShape(const CXXRecordDecl &RD, bool &HasGet,
                                 ->getCanonicalTypeUnqualified();
         }
       }
-    }
+    } break;
```
Comment (ymand) [inline clang/lib/Analysis/FlowSensitive/SmartPointerAccessorCaching.cpp:74]: newline?

**Review by ymand — APPROVED** [review]

File: clang/lib/Analysis/FlowSensitive/SmartPointerAccessorCaching.cpp (hunk 1 above)
Comment (jvoung) [reply clang/lib/Analysis/FlowSensitive/SmartPointerAccessorCaching.cpp:74]: clang-format wants to put it on that same line =/

Comment (DavidSpickett) [reply clang/lib/Analysis/FlowSensitive/SmartPointerAccessorCaching.cpp:74]: Could put it inside the `case : { ...}` scope, it might be ok with that.

Comment (DavidSpickett) [reply clang/lib/Analysis/FlowSensitive/SmartPointerAccessorCaching.cpp:74]: On second thought, there are a ton of these in the codebase already. It's fine as is.

**Comment by DavidSpickett** [comment]
Thanks for the fix!

**Comment by jvoung** [comment]
> Thanks for the fix!

Thanks for helping merge! And sorry again about the breakage!

**Pull request merged**